storage/innobase/lock/lock0lock.cc

```diff
@@ -1983,20 +1983,26 @@ RecLock::add_to_waitq(const lock_t* wait_for, const lock_prdt_t* prdt)
 	gathering) transactions. The problem is that we don't currently
 	block them using the TrxInInnoDB() mechanism. */
 
+	bool	high_priority = trx_is_high_priority(m_trx);
 #ifdef WITH_WSREP
-	if (wsrep_on(m_trx->mysql_thd) &&
+
+    /* Allow high priority transactions to jump over the low priority locks and
+     * rollback them if Galera is not loaded. */
```

Comment:
High priority transactions also work with codership mysql, I based the 8.0 fix on that. We could use the same patch here.

Reply:
Hi @venkatesh-prasad-v , can you please check it out?

Reply:
Hi @dutow!! I and Kamil decided to handle HPTs as part of PXC-3295. So, we have reverted the changes done in lock0lock.cc and will work towards evaluating the backport of the 8.0 fix to 5.7.